src/posts/posts.service.ts

```diff
@@ -30,18 +32,21 @@ export class PostsService {
     ): Promise<PostEntity> {
         let createFileDto: FileDto;
         if (file) {
-            createFileDto = await this.filesService.uploadFile(file.buffer, file.originalname);
+            createFileDto = await this.filesService.uploadFile(file);
             createPostDto.imageKey = createFileDto.key;
         }
-        const post = await this.postsRepo.create({
-            ...createPostDto,
-            user: { id: user.id },
-        });
-        const createdPost = await this.postsRepo.save(post);
-        if (createFileDto) {
-            await this.filesService.createFile(createFileDto, createdPost.id);
+        try {
+            const post = await this.postsRepo.create({ ...createPostDto, user: { id: user.id } });
+            const createdPost = await this.postsRepo.save(post);
+            if (createFileDto) {
+                await this.filesService.createFile(createFileDto, createdPost.id);
+            }
+            return createdPost;
+        } catch (err) {
+            if (err) {
+                throw new BadRequestException('Failed to create post');
```

Comment:
Please create custom exceptions for each module. You can have `common/exceptions` directory where can be created separate directories for each module: `posts`, `users`, etc...

Please check custom exceptions in my repository.